### PR TITLE
Remove a lot of nix code.

### PR DESCRIPTION
**Pull request by angerman**
This moves this project into a bare minimum nix project.

Missing:
- [x] cicero
- [x] coverageReports
- [x] ~setup derivation (though don't think we need them)~ 

This includes the changes from #381

### CODE REVIEW

**Comment by angerman** [comment]
@lehins is anyone looking at the coverage reports? Is it worth spending time trying to make them all green, or is this not consumed anyway?

**Comment by angerman** [comment]
Seems to be green. Waiting for cicero green as well.

File: flake.nix [hunk 1]
```diff
@@ -1,42 +1,189 @@
...
-    };
+    haskellNix.url = "github:input-output-hk/haskell.nix";
+    nixpkgs.follows = "haskellNix/nixpkgs-unstable";
+    iohkNix.url = "github:input-output-hk/iohk-nix";
+    flake-utils.url = "github:hamishmack/flake-utils/hkm/nested-hydraJobs";
+
+    CHaP.url = "github:input-output-hk/cardano-haskell-packages?ref=repo";
+    CHaP.flake = false;
+
+    # ciciero
```
Comment (andreabedini) [inline flake.nix:11]:
```suggestion
    # cicero
```

File: flake.nix [hunk 2]
```diff
@@ -1,42 +1,189 @@
 {
   inputs = {
-    nixpkgs.follows = "std/nixpkgs";
-    std.url = github:divnix/std;
-    tullia = {
-      url = github:input-output-hk/tullia;
-      inputs.std.follows = "std";
-    };
+    haskellNix.url = "github:input-output-hk/haskell.nix";
```
Comment (andreabedini) [inline flake.nix:3]: maybe worth sharing `tullia`?

```suggestion
    haskellNix.url = "github:input-output-hk/haskell.nix";
    haskellNix.inputs.tullia.follows = "tullia";
```

File: flake.nix [hunk 3]
```diff
@@ -1,42 +1,189 @@
...
+    inputs.flake-utils.lib.eachSystem supportedSystems (system:
+      let
+        # setup our nixpkgs with the haskell.nix overlays, and the iohk-nix
+        # overlays...
+        pkgs = import inputs.nixpkgs {
+          overlays = [inputs.haskellNix.overlay] ++ builtins.attrValues inputs.iohkNix.overlays;
+          inherit system;
+          inherit (inputs.haskellNix) config;
+        };
+        nixpkgs = pkgs;
```
Comment (andreabedini) [inline flake.nix:33]: why two names? I see you only reference nixpkgs at line 91 but that could use `pkgs` instead, or I am missing something?

File: flake.nix [hunk 4]
```diff
@@ -1,42 +1,189 @@
...
+
+          # tools we want in our shell
+          shell.tools = {
+            cabal = "3.10.1.0";
+            ghcid = "0.8.8";
+            haskell-language-server = "latest";
+            hlint = {};
+            weeder = "2.4.1";
+          };
+          # Not we use pkgsBuildBuild, to make sure that even in the cross
```
Comment (andreabedini) [inline flake.nix:59]: Now?
```suggestion
          # Now we use pkgsBuildBuild, to make sure that even in the cross
```

File: flake.nix [hunk 5]
```diff
@@ -1,42 +1,189 @@
...
+              # libsodium instead of the one from nixpkgs, which is what haskell.nix would
+              # otherwise choose. Unfortunately, this means that we also override any other
+              # pkgconfig libraries that haskell.nix would pick for us. So we also need to
+              # manually include those here.
+              packages.cardano-crypto-class.components.library.pkgconfig = pkgs.lib.mkForce [[ pkgs.libsodium-vrf pkgs.secp256k1 pkgs.blst ]];
+              packages.cardano-crypto-praos.components.library.pkgconfig = pkgs.lib.mkForce [[ pkgs.libsodium-vrf ]];
+              packages.slotting.configureFlags = [ "--ghc-option=-Werror" ];
+              enableLibraryProfiling = profiling;
+            })
+            ({pkgs, ...}: with pkgs; nixpkgs.lib.mkIf stdenv.hostPlatform.isWindows {
```
Comment (andreabedini) [inline flake.nix:91]:
```suggestion
            ({pkgs, ...}: with pkgs; lib.mkIf stdenv.hostPlatform.isWindows {
```

File: flake.nix (hunk 2 above)
Comment (angerman) [reply flake.nix:3]: I guess. thought we could also use tullia from haskell.nix?

**Review by andreabedini — COMMENTED** [review]
:+1:

File: flake.nix (hunk 3 above)
Comment (angerman) [reply flake.nix:34]: can't use pkgs there :-/ It's shadowed by another pkgs.

File: flake.nix (hunk 5 above)
Comment (angerman) [reply flake.nix:91]: this one can't come from pkgs here. as that leads to infinite recursion.

**Review by lehins — APPROVED** [review]
@angerman I am not gonna pretend to understand what's exactly going on here. I trust your experitse on this.

**Comment by angerman** [comment]
> @angerman I am not gonna pretend to understand what's exactly going on here. I trust your experitse on this.

There is only one file left 😂

I should try to document that one file even more. But that will have to wait until the 14th.

File: cabal.project [hunk 6]
```diff
@@ -9,9 +9,6 @@ repository cardano-haskell-packages
     c00aae8461a256275598500ea0e187588c35a5d5d7454fb57eac18d9edb86a56
     d4a35cd3121aa00d18544bb0ac01c3e1691d618f462c46129271bccf39f7e8ee
 
--- The hackage index-state is repeated twice to work around haskell.nix
--- limited parsing ability
-index-state: 2022-10-07T01:58:16Z
```
[the remark this answers is not in the document]
Comment (angerman) [reply cabal.project:14]: Well, I wouldn't call it fixed; the logic was removed. So it's not necessary anymore.

[the remark this answers is not in the document]
Comment (lehins) [reply cabal.project:14]: Thank you!

**Pull request merged**